File: searchRecursive.java
import java.util.Scanner;

public class searchRecursive {

    public static boolean isInArray(int[] arr, int n, int m) {
        if (n == 0) {
            return false;
        }
        if (arr[n - 1] == m) {
            return true;
        }
        return isInArray(arr, n - 1, m);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter size of an array:");
        int n = sc.nextInt();

        int[] arr = new int[n];
        System.out.println("Enter array elements:");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }

        System.out.println("Enter element to search:");
        int m = sc.nextInt();

        if (isInArray(arr, n, m)) {
            System.out.println("Element is present in array");
        } else {
            System.out.println("Element is not present in array");
        }

        sc.close();
    }
}

File: SearchRecursive.c
#include <stdio.h>

int isInarray(int arr, int n, int m) {
    if (n == 0)     return 0;
    if (arr[n - 1] == m)    return 1;
    
    return isInarray(arr, n - 1, m);
}

void main(){
    int n, m;
    printf("Enter size of an array: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter array elements:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    printf("Enter element to search: ");
    scanf("%d", &m);

    if (isInarray(arr, n, m)) {
        printf("Element is present in array");
    } else {
        printf("Element is not present in array");
    }
}